tests/lib/tools/snapd-state

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash -e
+
+show_help() {
+    echo "usage: check-state <jq-filter> <comparison> <expected-res> <error-message>"
+    echo "       print-state <jq-filter>"
+    echo "       change-snap-channel <snap-name> <channel>"
+    echo "       force-autorefresh"
+    echo "       prevent-autorefresh"
+    echo "       wait-for-autorefresh <last-change-id>"
+    echo ""
+    echo "The tool is used to manage the snapd state by editing the state.json file."
+    echo "Also it is used to check state and refresh status"
+}
+
+print_state() {
+    JQ_FILTER=$1
+    if [ -z "$JQ_FILTER" ]; then
+        echo "snapd-state: jq filter is needed"
+        exit 1
+    fi
+    # we are storing jq filter specs in JQFILTER, so we can't use --arg like 
+    # normal with jq, so instead just do a _real_ simple and potentially 
+    # _really_ annoying string concatenation with the actual verbatim variable
+    jq -r "$JQ_FILTER" < /var/lib/snapd/state.json
```

Comment:
How is this going to work with some more fancy jq invocations we have in the tests? Do we need $@?

Some random samples from our tests:
`MARKSEEDED_WAIT=$(jq -r '.tasks[] | select (.kind == "mark-seeded") | ."wait-tasks"' < /var/lib/snapd/state.json)`

or 

`test "$( snap run  jq -r ".data.snaps.\"$A_SNAP\".channel" < /var/lib/snapd/state.json )" = "$A_TRACK/candidate"`

in the latter we seem to be forcing jq from the snap for some reason, not sure why.

Reply:
I think it sholud work like this:
MARKSEEDED_WAIT=$("$TESTSTOOLS"/snapd-state print-state '.tasks[] | select (.kind == "mark-seeded") | ."wait-tasks"')"

In fact the filters used in disabled-svcs.sh are pretty fancy as well.